code/modules/mapping/map_template.dm

```diff
@@ -15,16 +18,39 @@
 	if(rename)
 		name = rename
 
-/datum/map_template/proc/preload_size(path, cache = FALSE)
+/datum/map_template/Destroy()
+	QDEL_NULL(cached_map)
+	return ..()
+
+/datum/map_template/proc/preload_size(path = mappath, force_cache = FALSE)
+	if(cached_map)
+		return cached_map.parsed_bounds
 	var/datum/parsed_map/parsed = new(file(path))
-	var/bounds = parsed?.bounds
```

Comment:
preload size usually deals with parsed bounds as it's done before it's actually loaded, added helper procs for getting the actual size as a list rather than bounds too.